proposed/http-client/http-client-meta.md

```diff
@@ -33,6 +33,11 @@ client will allow libraries to be decoupled from an implementation such as Guzzl
 specify the default behaviours.
 * The purpose is not to be opinionated about the use of middlewares (PSR-15).
 
+#### Asynchronous HTTP client
+
+The reason asynchronous support is a non-goal is because a PSR for HTTP clients 
+should not define Promises. That should be a separate PSR. Waiting for a such PSR 
```

Comment:
should define neither promises nor an async-compatible version of PSR-7

Reply:
I agree with @Nyholm here, the text LGTM and the discussion whether PSR-7 can be used in an async context is both an implementation detail and out of scope here :+1:

<small>(@kelunik IMO the last link adds little value here and its motivation is better described in the documentation (https://github.com/reactphp/http#request), rather than the code. As an alternative, PSR-7 does not prevent you from buffering the whole message in memory, in which case all accessors can be used perfectly fine even in an async program :+1:)</small>

Reply:
@clue It doesn't really matter as an async client is out of scope in any way, regardless of the wording here.

I completely understand the motivation behind that, but it doesn't really matter either. If it could be implemented sanely, I'm pretty sure you guys would have done that. It doesn't prevent buffering, of course not, but we both know that it doesn't fit for all bodies and is the exact reason why the request body isn't represented as a string, but a stream.